cnf-certification-test/networking/suite.go

```diff
@@ -259,3 +266,58 @@ func testDualStackServices(env *provider.TestEnvironment) {
 		ginkgo.Fail(fmt.Sprintf("Found %d services in error, check error log for details", len(errorServices)))
 	}
 }
+
+//nolint:funlen
+func testNetworkPolicyDenyAll(env *provider.TestEnvironment) {
+	ginkgo.By("Test for Deny All in network policies")
+	var namespacesMissingDenyAllDefaultPolicies []string
+	var processedNamespaces []string
+
+	for _, put := range env.Pods {
+		denyAllEgressFound := false
+		denyAllIngressFound := false
+
+		// Short circuit if the namespace has already been processed
+		if stringhelper.StringInSlice(processedNamespaces, put.Data.Namespace, false) {
+			logrus.Debugf("Namespace: %s has already been processed for deny-all network policies.  Skipping.", put.Data.Namespace)
+			break
+		}
+
+		// Look through all of the network policies for a matching namespace.
+		for index := range env.NetworkPolicies {
+			// Match the pod namespace with the network policy namespace.
+			if put.Data.Namespace == env.NetworkPolicies[index].Namespace {
+				// Check to see if the network policy is deny-all (and contains ingress, egress, (or both) network policy types)
+				denyAllExists, foundPolicies := policies.IsNetworkPolicyDenyAll(&env.NetworkPolicies[index])
+
+				if denyAllExists {
+					// Look through the returned policies to make sure they include both ingress and egress.
+					for _, p := range foundPolicies {
+						if p == v1.PolicyTypeEgress {
+							denyAllEgressFound = true
+						}
+
+						if p == v1.PolicyTypeIngress {
+							denyAllIngressFound = true
+						}
+					}
+				}
+			}
+		}
+
+		// Network policy has not been found that contains a deny-all rule for both ingress and egress.
+		if !denyAllIngressFound || !denyAllEgressFound {
+			namespacesMissingDenyAllDefaultPolicies = append(namespacesMissingDenyAllDefaultPolicies, put.Data.Namespace)
+			tnf.ClaimFilePrintf("Namespace %s was found to not have a default deny-all network policy.", put.Data.Namespace)
```

Comment:
Add which default policies are missing in the log, either ingress, egress or both

Reply:
Completed.